cmake/coroutineOptions.cmake

```diff
@@ -5,8 +5,6 @@ function(target_coroutine_options TARGET)
   if(MSVC)
     target_compile_options(${TARGET} PUBLIC /std:c++latest /permissive-)
   elseif(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
-    target_compile_options(${TARGET} PUBLIC -stdlib=libc++ -fcoroutines-ts)
-    target_link_options(${TARGET} PUBLIC -stdlib=libc++)
```

Comment:
I think `-fcoroutines-ts` is unnecessary since CMake should add it via `target_compile_features(concurrencpp PUBLIC cxx_std_20)` so I also removed it here.

Reply:
Well, it's more like providing flags for specific compilers should be the user's responsibility. Iirc I just copied this bit from what was there previously without major changes.

CMake will not add `-fcoroutines-ts`, the user will have to provide flags according to their own compiler, for which CMake provides many many opportunities.

Reply:
Yes, I think you are right. I think `-fcoroutines-ts` is only necessary on certain older compiler versions where coroutine support was still experimental. Specifying it for newer versions does not seem to hurt, so we could keep it or we can remove it (with the chance of breaking those older versions).